pkg/resource/json.go

```diff
@@ -0,0 +1,183 @@
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// JSONData provides a way to interact with a json data as a resource
+type JSONData struct {
+	// Provide common helper functions
+	BaseResource
+	raw       string
+	data      any
+	clonePath string
+	options   *JSONDataOptions
+}
+
+// JSONDataOptions are options for the JSONData resource
+type JSONDataOptions struct {
+	// Currently none needed but here for future cases
+}
+
+// NewJSONData returns a configured git repo resource for the scanner to scan
+func NewJSONData(raw string, options *JSONDataOptions) *JSONData {
+	return &JSONData{
+		raw:     raw,
+		options: options,
+	}
+}
+
+// Kind of resource (always returns JSONData here)
+func (r *JSONData) Kind() string {
+	return "JSONData"
+}
+
+// String representation of the resource
+func (r *JSONData) String() string {
+	return r.raw
+}
+
+// Clone the resource to the desired local location and store the path
+func (r *JSONData) Clone(path string) error {
+	r.clonePath = path
+	if err := os.MkdirAll(r.clonePath, 0700); err != nil {
+		return err
+	}
+
+	// Load the raw json into the data variable
+	if err := json.Unmarshal([]byte(r.raw), &r.data); err != nil {
+		return err
+	}
+
+	// Support droping specific files in the "repo" to configure scanner backends
+	for _, file := range []string{
+		".gitleaks.toml",
+		".gitleaksignore",
+		".gitleaksbaseline",
+	} {
+		if data, err := r.ReadFile(file); err == nil {
+			if err := os.WriteFile(filepath.Join(r.clonePath, file), data, 0600); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// ClonePath returns where this repo has been cloned if cloned else ""
+func (r *JSONData) ClonePath() string {
+	return r.clonePath
+}
+
+// Depth returns the depth for things that have version control
+func (r *JSONData) Depth() uint16 {
+	return 0
+}
+
+// SetDepth allows you to adjust the depth for the resource
+func (r *JSONData) SetDepth(depth uint16) {
+	// no-op
+}
+
+// SetCloneTimeout lets you adjust the timeout before the clone aborts
+func (r *JSONData) SetCloneTimeout(timeout time.Duration) {
+	// no-op
+}
+
+// Since returns the date after which things should be scanned for things
+// that have versions
+func (r *JSONData) Since() string {
+	return ""
+}
+
+// ReadFile provides a way to access values in the JSON data
+func (r *JSONData) ReadFile(path string) ([]byte, error) {
+	// Build out the path components
+	var components []string
+	for _, component := range strings.Split(path, string(filepath.Separator)) {
```

Comment:
This is problematic for cross platform. 
This is all in memory, right? If that is the case should we standardize on the representation within the json being "unix"?

Reply:
I want `ReadFile` to be consistent across all the different resource types and work with the output of [filepath.Join](https://pkg.go.dev/path/filepath#Join). Since it joins elements with an OS specific [Separator](https://pkg.go.dev/path/filepath#Separator) I wanted to make sure they lined up.

So in theory this should make it less problematic cross platform, right?

Reply:
Sounds reasonable. 
We can use `filepath.join` to create `path` entries in the tests. I've sent a diff separately.